### ISSUE TITLE
Documentation Related Requests & Decisions

### ISSUE THREAD

**Issue by ChristopherA** [ChatTheatre/eOS#2]
Documentation Related Requests and Decisions for @noahgibbs 

* [x] Contract, Copyright and License Related
  * [x] Re-sign Cycle 2020-E work order
  * [x] Add CC0 declaration to
    * [x] https://github.com/ChatTheatre/eOS-Doc/blob/master/README.md
    * [x] https://github.com/ChatTheatre/SkotOS-Doc/blob/master/README.md
    * [x] https://github.com/ChatTheatre/kernellib-doc/blob/master/README.md
    * [x] https://github.com/ChatTheatre/kernellib-doc/blob/master/README.md
    * [x] https://github.com/ChatTheatre/lpc-doc/blob/master/README.md
  * [x] Add CLA.md and ./CLA-signed/cla*.asc (and AGPL `LICENSE` from file if needed)
      * [x] https://github.com/ChatTheatre/SkotOS/
      * [x] https://github.com/ChatTheatre/eOS/
      * [x] https://github.com/ChatTheatre/thin-auth
      * [x] https://github.com/ChatTheatre/orchil
      * [x] https://github.com/ChatTheatre/websocket-to-tcp-tunnel

Note that with all of the following I've create github webpage for, so after edit pass confirm that the webpages look correct.

* [x] Delete your repo and test install instructions for https://github.com/ChatTheatre/eOS/ and update README.md accordingly

* [x] Document what each of the folders are, and what for, in https://github.com/ChatTheatre/eOS/blob/master/README.md

* [x] Review docs at bottom of https://github.com/ChatTheatre/kernellib/blob/master/README.md and edit/move docs as appropriate to https://github.com/ChatTheatre/kernellib-doc/ instead.
  * [x] In https://github.com/ChatTheatre/kernellib-doc/  add TOC to README.md and one-two line abstract for each doc moved there.
  * [x] Set up framework for same TOC style in other -Docs repo README.md

* [x] Edit/Merge to eOS-doc, name appropriately (eOS-architecture?)  and delete originals (if not in SkotOS they may have been moved to SkotOS-Docs)
  * https://github.com/ChatTheatre/SkotOS/blob/master/docs/layers.md
  * https://github.com/ChatTheatre/SkotOS/blob/master/docs/dgd/layers.md

* [ ] Choose and document an initial strategy for how we want to add examples to /usr
  * [x] Should we create empty ./usr/ folder in eOS repo?
  * [ ] Should there be one or more examples that are under eOS repo control?
  * [ ] What is MVP vs a mimimal plugin strategy.
  * [ ] Or do we tell people to fork examples /usr/ into their github folders, submodule them in their personal repos, and then we have .gitignore for /usr/* in the eOS repo so that they don't get picked up?
  * [ ] Or do we tell people to `wget` or `rsync -rlp --exclude '.git' user@host:path/to/git/repo/ .` contents of a example repo, such as Nino's example, and tell them to `git init` it to their personal github accounts?

* [ ] Create a prioritized list of legacy DGD / LPC / kernellib and other documents to convert/update for eOS, and list them in a eOS markdown file with pointer to source file.

**Comment by noahgibbs** [comment]
Kernellib-doc is listed twice in the first section.

lpc-doc appears to be one of Dworkin's repositories, like kernellib. Do we want to add more to it, or do we consider the existing disclaimers adequate (as we do with kernellib)?

**Comment by noahgibbs** [comment]
We definitely want a /usr directory in eOS. It's necessary as a location for linking other modules (mini-libraries) once they're properly broken up into repositories.

My planned MVP for modules is to update the "modules" files used by SkotOS a bit to include information on where to git-clone them, not just the names of modules being depended on. Then an external script running outside DGD can clone the repositories and link them into appropriate places, normally inside the usr directory. The same thing can be done with submodules, either by the script or by a downstream user.

I don't think that having people effectively copy repos into place is a good idea. It can make for a really ugly update story later if they want to incorporate newer changes. The MVP modules system isn't perfect either, but it's better than that.

**Comment by ChristopherA** [comment]
> Do we want to add more to it, or do we consider the existing disclaimers adequate (as we do with kernellib)?

If there are any good LPC specific resources that support or teach the language that are kernel and game indepedent, we should add them here as long as the are public domain and CC0 like these files from Dworkin are. If it do a good job he might accept a PR back to his root repo. If not, it will be useful for our people.

**Comment by ChristopherA** [comment]
> We definitely want a /usr directory in eOS. It's necessary as a location for linking other modules (mini-libraries) once they're properly broken up into repositories.

I added `./usr` and an empty `./usr/.gitignore` in https://github.com/ChatTheatre/eOS/commit/a5bb611105fff939ece9532396b0bbd44d909446 and changed `./usr/.gitignore` to ignore the default `./usr/admin` directory created with the first user logs in in https://github.com/ChatTheatre/eOS/commit/de7a2f9070db4c909563c9c4d2b9c9b4673fb498.

**Comment by noahgibbs** [comment]
If the plan is that we want to get as much SkotOS functionality as possible and eOS is significantly in second place, I assume that means I should be moving most SkotOS-type documentation into SkotOS-Doc rather than eOS-Doc.

I'm thinking specifically of the item called "Edit/Merge to eOS-doc, name appropriately (eOS-architecture?) and delete originals (if not in SkotOS they may have been moved to SkotOS-Docs)".

Presumably we *want* those to be in SkotOS-Docs, while eOS will probably slowly add links to relatively low-level code like the Kernellib, etc.

**Comment by ChristopherA** [comment]
I agree that those items relevant to both SkotOS and eOS should be in SkotOS-Docs, with the exception of generic (not SkotOS specific) kernellib related items which should be in KernelLib-Docs.

**Comment by noahgibbs** [comment]
Sounds good. Agree.
